IOPool/Input/src/RootEmbeddedFileSequence.cc

```diff
@@ -43,7 +44,8 @@ namespace edm {
         initialNumberOfEventsToSkip_(pset.getUntrackedParameter<unsigned int>("skipEvents", 0U)),
         treeCacheSize_(pset.getUntrackedParameter<unsigned int>("cacheSize", roottree::defaultCacheSize)),
         enablePrefetching_(false),
-        enforceGUIDInFileName_(pset.getUntrackedParameter<bool>("enforceGUIDInFileName", false)) {
+        enforceGUIDInFileName_(pset.getUntrackedParameter<bool>("enforceGUIDInFileName", false)),
+        fileOpenAttempts_(pset.getUntrackedParameter<unsigned int>("fileOpenAttempts", numberOfFiles())) {
```

Comment:
In https://github.com/cms-sw/cmssw/issues/15653 you wrote
> In addition, using the number of files in the list as the retry count doesn't make much sense for a random selection, and also doesn't make sense if retries are being cause by a systemic problem like a site issue.

Should the default be something else then? The `fillDescriptions()` (which, IIUC, is not used by any of the mixing modules) uses `1`. Should we use the same here (also to not change the default behavior)?

Reply:
@makortel `fileOpenAttempts` is new in this PR.  The old behavior was to try up to the number of files on the initial open, and then not retry at all for subsequent files.  Rereading the HN thread, `numberOfFiles()` is probably too large, but 1 is too small--1 would be equivalent to turning off `skipBadFiles`.  I also realized rereading the HN thread that what @bbockelm was suggesting was a per-job retry limit, so that systemic problems don't result in the servers getting hammered by retries from every thread--so I've revised the PR to set the retry count to 3 and make it a static atomic.